Tests for MAP transform failing

For the MAP transform binding, the tests are failing.
```
$ mvn clean install -DskipChecks
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building openHAB Add-ons :: Bundles :: Transformation Service :: Map 2.5.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-clean-plugin:3.0.0:clean (default-clean) @ org.openhab.transform.map ---
[INFO] Deleting D:\dev\openhab2\git\openhab2-addons\bundles\org.openhab.transform.map\target
[INFO]
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-java) @ org.openhab.transform.map ---
[INFO]
[INFO] --- directory-maven-plugin:0.3.1:directory-of (directories) @ org.openhab.transform.map ---
[INFO] Directory of org.openhab.addons:org.openhab.addons.reactor set to: D:\dev\openhab2\git\openhab2-addons
[INFO]
[INFO] --- karaf-maven-plugin:4.2.6:features-generate-descriptor (compile) @ org.openhab.transform.map ---
[INFO] Generating feature descriptor file D:\dev\openhab2\git\openhab2-addons\bundles\org.openhab.transform.map\target\feature\feature.xml
[INFO]  Tree listing is saved here: D:\dev\openhab2\git\openhab2-addons\bundles\org.openhab.transform.map\target\history\treeListing.txt

[INFO] ...done!
[INFO] Attaching features XML
[INFO]
[INFO] --- maven-resources-plugin:3.0.2:resources (default-resources) @ org.openhab.transform.map ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ org.openhab.transform.map ---
[INFO] Changes detected - recompiling the module!
[INFO]
[INFO] --- bnd-maven-plugin:4.2.0:bnd-process (default) @ org.openhab.transform.map ---
[WARNING] D:\dev\openhab2\git\openhab2-addons\bundles\org.openhab.transform.map\pom.xml [0:0]: Unused Export-Package instructions: [org.openhab.*]
[WARNING] D:\dev\openhab2\git\openhab2-addons\bundles\org.openhab.transform.map\pom.xml [0:0]: Unused Import-Package instructions: [io.swagger.annotations.*, javax.annotation.security.*, org.eclipse.jdt.annotation.*, org.openhab.core.automation.annotation.*, org.openhab.*, com.google.common.*]
[INFO]
[INFO] --- maven-resources-plugin:3.0.2:testResources (default-testResources) @ org.openhab.transform.map ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory D:\dev\openhab2\git\openhab2-addons\bundles\org.openhab.transform.map\src\test\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ org.openhab.transform.map ---
[INFO] Changes detected - recompiling the module!
[INFO]
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ org.openhab.transform.map ---
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.openhab.transform.map.internal.MapTransformationServiceTest
[main] WARN org.eclipse.smarthome.core.transform.AbstractFileTransformationService - Could not transform 'toBeDefaulted' with the file 'map/doorstatus.map' : Target value not found in map for 'toBeDefaulted'
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.062 s <<< FAILURE! - in org.openhab.transform.map.internal.MapTransformationServiceTest
[ERROR] testTransformByMap(org.openhab.transform.map.internal.MapTransformationServiceTest)  Time elapsed: 0 s  <<< ERROR!
java.io.IOException: Unable to delete directory target\conf\transform.
        at org.openhab.transform.map.internal.MapTransformationServiceTest.init(MapTransformationServiceTest.java:62)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   MapTransformationServiceTest.init:62 ▒ IO Unable to delete directory target\co...
[INFO]
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.810 s
[INFO] Finished at: 2019-07-22T19:42:32+02:00
[INFO] Final Memory: 33M/658M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project org.openhab.transform.map: There are test failures.
[ERROR]
[ERROR] Please refer to D:\dev\openhab2\git\openhab2-addons\bundles\org.openhab.transform.map\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

--- a/bundles/org.openhab.transform.map/src/test/java/org/openhab/transform/map/internal/MapTransformationServiceTest.java
+++ b/bundles/org.openhab.transform.map/src/test/java/org/openhab/transform/map/internal/MapTransformationServiceTest.java
@@ -23,10 +23,13 @@ import java.util.Properties;
 import java.util.concurrent.Callable;
 
 import org.apache.commons.io.FileUtils;
+import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
-import org.openhab.transform.map.internal.MapTransformationService;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.osgi.framework.BundleContext;
 
 /**
  * @author Gaël L'hopital
@@ -44,23 +47,46 @@ public class MapTransformationServiceTest {
     private static final String CONFIG_FOLDER = BASE_FOLDER + File.separator + SRC_FOLDER;
     private static final String USED_FILENAME = CONFIG_FOLDER + File.separator + "transform/" + EXISTING_FILENAME_DE;
 
-    private MapTransformationService processor;
+    @Mock
+    private BundleContext bundleContext;
+
+    private TestableMapTransformationService processor;
+
+    private class TestableMapTransformationService extends MapTransformationService {
+        @Override
+        protected String getSourcePath() {
+            return BASE_FOLDER + File.separator + super.getSourcePath();
+        }
+
+        @Override
+        protected Locale getLocale() {
+            return Locale.US;
+        }
+
+        @Override
+        public void activate(BundleContext context) {
+            super.activate(context);
+        }
+
+        @Override
+        public void deactivate() {
+            super.deactivate();
+        }
+    };
 
     @Before
-    public void init() throws IOException {
-        processor = new MapTransformationService() {
-            @Override
-            protected String getSourcePath() {
-                return BASE_FOLDER + File.separator + super.getSourcePath();
-            }
+    public void setUp() throws IOException {
+        MockitoAnnotations.initMocks(this);
 
-            @Override
-            protected Locale getLocale() {
-                return Locale.US;
-            }
-        };
-        FileUtils.deleteDirectory(new File(CONFIG_FOLDER));
+        processor = new TestableMapTransformationService();
+        processor.activate(bundleContext);
         FileUtils.copyDirectory(new File(SRC_FOLDER), new File(CONFIG_FOLDER));
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        processor.deactivate();
+        FileUtils.deleteDirectory(new File(CONFIG_FOLDER));
     }
 
     @Test

